fix: review content in Grammarly (#285)
nhoizey/nicolas-hoizey.com: src/articles/2020/05/05/jamstack-is-fast-only-if-you-make-it-so/index.md

# Enhance client-side, if really needed

If you know you receive a lot of very useful webmentions that you have to show to your visitors, you can enhance the server-side generated list with a bit of client-side.

But remember every JavaScript added to the page has a cost, so the few additional webmentions have to be really useful.

So, instead of doing this for every page view, at least:

First, try to **wait for some time after the site build** before making client-side API calls. Keep the build timestamp available to client-side JavaScript, and wait for an hour, a day, or more, depending on the frequency of webmentions. You could even use the page's "age" to query webmention.io less for older content that probably receives less webmentions, as [Aaron Gustafson did even for server-side call in his Jekyll plugin](https://aarongustafson.github.io/jekyll-webmention_io/performance-tuning).

Then, **keep track of a user's calls to the API**, in localStorage or IndexedDB, so that you don't make these calls again a short while after. You could even use a Service Worker to cache requests and their timestamp.

# Client-side only API calls sometimes make more sense

> I agree Webmentions are not the most complex use case to explain that you should use as much as possible call APIs from the server at build time instead of from the client:

Author: ```suggestion
I agree Webmentions are not the most complex use case to explain that most of the time you should call APIs from the server at build time rather than from the client:
```

![JAMstack layers enhanced](wtf-jamstack-layers-enhanced.png)

It makes it more obvious there is a pile of things (quite useful for a "stack") and JavaScript is the link between APIs and Markup.

Of course, it reads as AJMstack instead of JAMstack, so I bet I won't be successful promoting it, but at least it feels more accurate.

> It even allows presenting this as a significant [progressive enhancement](/tags/progressive-enhancement/) platform, as we can start with plain old (did I hear "boring"?) Markup…

Reviewer: I would keep "great" here.

> It even allows presenting this as a significant [progressive enhancement](/tags/progressive-enhancement/) platform, as we can start with plain old (did I hear "boring"?) Markup…

Author: ```suggestion
It even allows presenting this as a great [progressive enhancement](/tags/progressive-enhancement/) platform, as we can start with plain old (did I hear "boring"?) Markup…
```
